### DISCUSSION TITLE
Instead of disabling a rule inline in a file, make it a warning

### DISCUSSION THREAD

**Discussion by cafesanu** [eslint/eslint#18099] [Ideas]
Hi, so I use eslint and vscode. We have many eslint rules that have been disabled inline via `// eslint-disable-next-line`. Now, sometimes disabling a rule makes a lot of sense depending on the situation. However, many times it doesn't, and disabling completely hides the issue forever. Your IDE or CI won't tell you that there might be something wrong with a line of code. Forever. Unless you stumble upon the comment when you open the file.

Another issue is that, when disabling a rule, neither `eslint -fix` nor vscode or any other IDE will give you and option to quick fix it.

I would love to see a new addition where you can convert an error to a warning inline, this way, github PRs will show the warning as well as `eslint --fix` or your IDE will try to fix it via `quick fix`.

Wondering if this is possible.

Thanks!

**Comment by Rec0iL99** [comment]
Hi @cafesanu, this is possible by overriding the rule options using [configuration comments](https://eslint.org/docs/next/use/configure/rules#using-configuration-comments).

```js
// eslint.config.js
export default [
  {
    rules: {
      quotes: ["error", "double"],
    },
  },
];
```

```js
/* eslint quotes: ["warn"] */
const foo = 'bar';
```

By using configuration comments you are now able to override the severity of the rule from `error` to `warn` and using the `--fix` CLI option should also work.

**Reply by cafesanu** [reply]
Nice @Rec0iL99 but this is for the whole file right? It would be awesome if you could do it per line, just like `eslint-disable-next-line`

**Reply by Rec0iL99** [reply]
@cafesanu feel free to open a new feature request providing all details and use cases.
